File: app/src/main/java/com/bytedance/tiktok/adapter/ShareAdapter.java
package com.bytedance.tiktok.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import com.bytedance.tiktok.R;
import com.bytedance.tiktok.base.BaseRvAdapter;
import com.bytedance.tiktok.base.BaseRvViewHolder;
import com.bytedance.tiktok.bean.ShareBean;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * create by libo
 * create on 2020-05-25
 * description
 */
public class ShareAdapter extends BaseRvAdapter<ShareBean, ShareAdapter.ShareViewHolder> {

    public ShareAdapter(Context context, List<ShareBean> datas) {
        super(context, (ArrayList<ShareBean>) datas);
    }

    @Override
    protected void onBindData(ShareViewHolder holder, ShareBean shareBean, int position) {
        holder.tvIcon.setText(shareBean.getIconRes());
        holder.tvText.setText(shareBean.getText());
        holder.viewBg.setBackgroundResource(shareBean.getBgRes());
    }

    @NonNull
    @Override
    public ShareViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(getContext()).inflate(R.layout.item_share, parent, false);
        return new ShareViewHolder(view);
    }

    public class ShareViewHolder extends BaseRvViewHolder {
        @BindView(R.id.tv_icon)
        TextView tvIcon;
        @BindView(R.id.tv_text)
        TextView tvText;
        @BindView(R.id.view_bg)
        View viewBg;

        public ShareViewHolder(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/bytedance/tiktok/adapter/ShareAdapter.kt
package com.bytedance.tiktok.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import com.bytedance.tiktok.adapter.ShareAdapter.ShareViewHolder
import com.bytedance.tiktok.base.BaseAdapter
import com.bytedance.tiktok.bean.ShareBean
import com.bytedance.tiktok.bean.VideoBean
import com.bytedance.tiktok.databinding.ItemShareBinding

/**
 * create by libo
 * create on 2020-05-25
 * description
 */
class ShareAdapter : BaseAdapter<ShareViewHolder, ShareBean>(ShareDiff()) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ShareViewHolder {
        return ShareViewHolder(ItemShareBinding.inflate(LayoutInflater.from(parent.context), parent, false))
    }

    override fun onBindViewHolder(holder: ShareViewHolder, position: Int) {
        var shareBean = mList[position]
        holder?.binding?.tvIcon!!.setText(shareBean!!.iconRes)
        holder?.binding?.tvText!!.text = shareBean?.text
        holder?.binding?.viewBg!!.setBackgroundResource(shareBean!!.bgRes)
    }

    inner class ShareViewHolder(val binding: ItemShareBinding) : RecyclerView.ViewHolder(binding.root) {

    }
}

class ShareDiff: DiffUtil.ItemCallback<ShareBean>() {
    override fun areItemsTheSame(oldItem: ShareBean, newItem: ShareBean): Boolean {
        return (oldItem.iconRes == newItem.iconRes)
    }

    override fun areContentsTheSame(oldItem: ShareBean, newItem: ShareBean): Boolean {
        return (oldItem.iconRes == newItem.iconRes && oldItem.text.equals(newItem.text))
    }

}